### PR TITLE
Fix: Player ready stuff

### PR DESCRIPTION
**Pull request by itschip**
Please come with some input.

### CODE REVIEW

**Comment by RockySouthpaw** [comment]
How exactly would you like this tested? I see there is esx code to be commented out. I can test on a non multichar/non esx build, plume and then plume + kash if these are all needed.

will be some hours though

File: resources/server/players/player.controller.ts [hunk 1]
```diff
@@ -52,6 +53,37 @@ if (!config.general.enableMultiChar) {
   });
 }
 
+//// comment out this if you're not using ESX, idk. Testing purpose
+
+if (config.general.enableMultiChar) {
+  on('onServerResourceStart', async (resource: string) => {
+    if (resource === GetCurrentResourceName()) {
+      // Workaround till https://github.com/citizenfx/fivem/pull/682
```
Comment (AvarianKnight) [inline resources/server/players/player.controller.ts:61]: Maybe do a export here (though I don't know if this is valid in TS/JS
```ts
const framework: {} = exp[config.framework.name].[config.framework.getter]()
const onlinePlayers: string[] = getPlayers()
for (const id of onlinePlayers) {
   const source: Number = parseInt(id)
   const Player = framework.[config.framework.playerGetter](parseInt(id))
   if (Player) {
     const identifier = Player.[config.framework.identifierGetter]()
     const firstname = Player.[config.framework.nameGetter]()
     await PlayerService.handleNewPlayerEvent({ source, identifier, firstname })
   }
}
```
```

**Review by RockySouthpaw — APPROVED** [review]
Tested with config set to false and on a non esx server. Worked fine.

Tested with config set to false and plume esx. Worked fine.

Tested with config set to true and plume esx. Worked fine.

File: resources/server/players/player.controller.ts [hunk 2]
```diff
@@ -52,6 +53,37 @@ if (!config.general.enableMultiChar) {
...
+      // is merged
+      // @ts-ignore
+      const onlinePlayers: XPlayer[] = ESX.GetPlayers();
```
[the remark this answers is not in the document]
Comment (itschip) [reply resources/server/players/player.controller.ts:64]: This is not going to be merged. Simply pushed for you to test.

**Comment by AvarianKnight** [comment]
All works fine 👍

**Pull request merged**